tests/utils.cpp

```diff
@@ -281,6 +274,83 @@ TEST(utils, sanitiseBPFProgramName)
             "is_a_very_long_function_name_which_exceeds_the_ba30ddc67a52bad2");
 }
 
+// Run a function with environment var set to specific value.
+// Does its best to clean up env var so it doesn't leak between tests.
+static void with_env(const std::string &key,
+                     const std::string &val,
+                     std::function<void()> fn)
+{
+  EXPECT_EQ(::setenv(key.c_str(), val.c_str(), 1), 0);
+  try
+  {
+    fn();
+  }
+  catch (const std::exception &ex)
+  {
+    EXPECT_EQ(::unsetenv(key.c_str()), 0);
```

Comment:
Why not unset even on success (perhaps just add this to the top in a scope exit?)

Reply:
Yeah, we're missing SCOPE_EXIT style macros in bpftrace. Should probably add it at somepoint